cartography/data/jobs/analysis/semgrep_sca_risk_analysis.json

```diff
@@ -0,0 +1,66 @@
+{
+    "name": "Semgrep SCA findings reachability risk analysis based on likelihood and impact. Impact = Severity, Likelihood = reachability + reachability_check",
+    "statements": [
+        {
+            "__comment__": "ARCHIVED -> Risk = Info",
+            "query": "MATCH (g:GitHubRepository{archived:true})<-[:FOUND_IN]-(s:SemgrepSCAFinding) SET s.reachability_risk = 'INFO' return COUNT(*) as TotalCompleted",
+            "iterative": false
+        },
+        {
+            "__comment__": "reachability = UNREACHABLE -> Risk = Info",
+            "query": "MATCH (s:SemgrepSCAFinding{reachability:'UNREACHABLE'}) SET s.reachability_risk = 'INFO' return COUNT(*) as TotalCompleted",
```

Comment:
since there is already a 1:1 mapping between finding and repo, it seems like all of these calculations can be done from python in the "transform" method.

Reply:
Is this a recommendation or a must?
TL;DR it's in cypher because that is what other module do.

I followed https://lyft.github.io/cartography/dev/writing-analysis-jobs.html?highlight=analysis and examples at https://github.com/lyft/cartography/blob/master/cartography/data/jobs/analysis to put any extra information based on an analysis, not from the original source, into the ingested nodes. The data is saved into the same node and as part of the same sync, but required an analysis to determine the new attribute.

If you have any other examples that follow that pattern (analyze, calculate and add data during transform step) I could follow them to keep the approach consistent in the codebase.

Reply:
I won't require it, but I think it will all be much more performant, understandable and testable if done in python, since it wouldn't need IO to the database.

Reply:
I really like analysis jobs because they let us simplify the graph in powerful ways, but they also scare me because they can update and remove fields in ways that are hard to understand if not done carefully. 

Converting an analysis job to run in Python code with a transform function with the data model gives us nice conveniences like automatic cleanups of stale attributes and relationships, but it's also inflexible and can marginally slow things down since all the IO will be handled by Python and not the database.

So I think we should keep things as an analysis job in this PR and ensure that we test for the right things and create the right patterns so that other analysis jobs can copy them.

Thanks for being patient as we figure this out.

Reply:
>I won't require it, but I think it will all be much more performant, understandable and testable if done in python, since it wouldn't need IO to the database.

Pretty sure that for perf it's generally better to keep things done by the DB and having data hit Python makes it slower. We are willing to take the perf hit though if it makes things more testable and maintainable (like with introducing the data model). However in this case I think being able to run neo4j queries to do analysis is useful and we should find ways to build guardrails around them.